dadi/lib/composer/index.js

```diff
@@ -89,6 +83,13 @@ Composer.prototype.composeOne = function (doc, callback) {
             }
           }
 
+          // Are we composing a media document? If so, we need to format it
```

Comment:
@eduardoboucas I don't particularly like it either. Internal hooks are a sound idea. Agree that's it's worth revisiting at some point!